contracts/test/RefinanceTest.t.sol

```diff
@@ -306,4 +306,170 @@ contract RefinanceTest is AddressRegistry, TestUtils {
+        assertEq(loanV3.drawableFunds(),       1_000_000_000000);
+        assertEq(loanV3.claimableFunds(),      0);
+
+        borrower.loan_drawdownFunds(address(loanV3), 1_000_000_000000, address(borrower));
+
+        vm.warp(start + 60 days);
+
+        ( uint256 principalPortion, uint256 interestPortion, uint256 delegateFee, uint256 treasuryFee ) = loanV3.getNextPaymentBreakdown();
+
+        assertEq(principalPortion, 0);
+        assertEq(interestPortion,  19_726_027397); // 2,000,000 * 12% * 30 / 365
+        assertEq(delegateFee,      410_958904);
+        assertEq(treasuryFee,      410_958904);
+
+        uint256 totalPayment = principalPortion + interestPortion + delegateFee + treasuryFee;
+
+        erc20_mint(USDC, 9, address(borrower), totalPayment);
+
+        borrower.erc20_approve(USDC, address(loanV3), totalPayment);
+        borrower.loan_makePayment(address(loanV3), totalPayment);
+
+        assertEq(loanV3.drawableFunds(),      0);
+        assertEq(loanV3.claimableFunds(),     interestPortion);
+        assertEq(loanV3.nextPaymentDueDate(), start + 90 days);
+        assertEq(loanV3.principal(),          2_000_000_000000);
+        assertEq(loanV3.paymentsRemaining(),  1);
+        
+        uint256[7] memory details = pool.claim(address(loanV3), address(debtLockerFactory));
+
+        assertEq(details[0], interestPortion);
+        assertEq(details[1], interestPortion);
+        assertEq(details[2], 0);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 0);
+        assertEq(details[6], 0);
+    }
+
+    function test_refinance_decreasedPrincipal() external {
+        bytes[] memory calls = new bytes[](2);
+        calls[0] = abi.encodeWithSelector(Refinancer.setEndingPrincipal.selector, 750_000_000000);
+        calls[1] = abi.encodeWithSelector(Refinancer.decreasePrincipal.selector,  250_000_000000);
+
+        borrower.loan_proposeNewTerms(address(loanV3), address(refinancer), deadline, calls);
+
+        erc20_mint(USDC, 9, address(borrower), 250_000_000000);
+
+        borrower.erc20_approve(USDC, address(loanV3), 250_000_000000);
+        borrower.loan_returnFunds(address(loanV3), 250_000_000000);
+
+        DebtLocker(loanV3.lender()).acceptNewTerms(address(refinancer), deadline, calls, 0);
+
+        assertEq(loanV3.principal(),           750_000_000000);
+        assertEq(loanV3.endingPrincipal(),     750_000_000000);
+        assertEq(loanV3.interestRate(),        0.12e18);
+        assertEq(loanV3.paymentsRemaining(),   2);
+        assertEq(loanV3.paymentInterval(),     30 days);
+        assertEq(loanV3.delegateFee(),         154_109589);  // 750,000 * 0.5% * 30 / 365
+        assertEq(loanV3.treasuryFee(),         154_109589);
+        assertEq(loanV3.refinanceCommitment(), 0);
+        assertEq(loanV3.drawableFunds(),       0);
+        assertEq(loanV3.claimableFunds(),      250_000_000000);
+
+        pool.claim(address(loanV3), address(debtLockerFactory));
+
```

Comment:
Need to assert all claim info as well as balance changes in loan and liquidity locker and `principalOut` change in pool.

Need to make sure that this claim properly accounts for the 250k principal that was returned by the borrower